doc/vxlan/Overlay ECMP ehancements.md

```diff
@@ -0,0 +1,227 @@
+- Once the last primary endpoint goes down, The route would switch to the secondary endpoint ECMP group based on BFD state. 
+- When running with secondary endpoints, if one primary comes back Up, the route nexthop would switch to primarynexthop group. 
+
+The following diagram depicts the scenario where Endpoint 1 and 2 are primary in a tunnel route and Endpoint 3 and 4 are backup. The active endpoints 1 and 2 are indicated with green line.
+
+![](https://github.com/sonic-net/SONiC/blob/22e06c87939f49ee72687cf2972f83a526c67b30/images/vxlan_hld/OverlayEcmp_priorty.png)
+
+
+### 3.1.1 Behaviour example
+
+| Scenario |     Before state change    |       State change       | Result                 | Description|
+|:--------:|:--------------------------:|:------------------------:|:----------------------:|:-----------|
+| Single primary-secondary switchover. Endpoint list = [A, A’], Primary[A] | NH=[A]  | A went Down  | NH=[A’]  | NH has a single primary endpoint which upon failing is replaced by the single Backup endpoint. |
+| Single primary recovery. Endpoint list = [A, A’] Primary[A] | NH=[A’]  | A is back up  | NH=[A]  | NH has a single backup endpoint which upon recovery of primary is replaced.  |
+| Single primary backup Failure. Endpoint list = [A, A’].  Primary[A]| NH=[A’] A is DOWN | A’ goes Down  | NH=[]  | No active Endpoint results in route being removed.  |
+| Multiple primary backups. Single primary failure.  Endpoint list = [A, B, C, A’, B’, C’]  Primary = [A, B, C] | NH = [A, B, C]  | A goes Down  | NH=[B, C]  | One of the primaries goes down. The others stay active.  |
+| Multiple primary backups. Multiple primary failure. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [B, C]  | B goes Down. A already Down. | NH=[C]  | 2 of the primaries goes down. The 3rd stay active. NH group is updated.  |
+| Multiple primary backups.  All primary failure. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [C]  | C goes Down. A,B already Down  | NH=[A’, B’, C’]  | All the primaries are down. The backup endpoints are added to the NH group.  |
+| Multiple primary backups. Backup Failure. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [A’, B’, C’]  | C’ goes Down. A, B, C already Down  | NH=[A’, B’]  | All the primaries are down. Failure of a backup endpoint shall result in its removal from NH.  |
+| Multiple primary backups. Single primary recovery. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [A’, B’, C’]  | A is Up. B, C still Down  | NH=[A]  | Primary takes precedence and is added to the NH. All the backups are removed.  |
+| Multiple primary backups. Multiple primary recovery. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [A]  | A is Up. B, C also come up  | NH=[A, B, C]  | Primary endpoints take precedence and are added to the NH.  |
+ 
+
+## 3.2 Custom monitoring for VTEP liveness detection.
```

Comment:
Can we capture the module interaction as well by enhancing the diagram [present in the ](https://github.com/sonic-net/SONiC/blob/master/doc/vxlan/Overlay%20ECMP%20with%20BFD.md#23-module-interaction)

Reply:
done. a new diagram has been added.